### ISSUE TITLE
Test Failure 

### ISSUE THREAD

**Issue by jakelandis** [logstash-plugins/logstash-input-beats#222]
More test failures unrelated to new code. 

```
2017/06/08 21:27:30.854710 Waiting for 1 prospectors to initialise
2017/06/08 21:27:30.854759 Launching harvester on new file: /tmp/studtmp-42b94bfe144419af867e5284ce7142a9e5e65a55c353eab089a1d5a0f07a
2017/06/08 21:27:30.854826 harvest: "/tmp/studtmp-42b94bfe144419af867e5284ce7142a9e5e65a55c353eab089a1d5a0f07a" (offset snapshot:0)
2017/06/08 21:27:30.854873 All prospectors initialised with 0 states to persist
2017/06/08 21:27:30.855226 Setting trusted CA from file: /tmp/studtmp-725b02522f1fc58ec885ec14696ecfb9e3347fce3e66c5506dca5d4e4fca
2017/06/08 21:27:30.856198 Connecting to [127.0.0.1]:44172 (localhost) 
2017/06/08 21:27:30.856328 Failure connecting to 127.0.0.1: dial tcp 127.0.0.1:44172: connection refused
). Sleeping for 2.0
Failed (Expected: 5 got: 0, execution output:
 2017/06/08 21:27:30.854316   --- options -------
2017/06/08 21:27:30.854361  config-arg:          /tmp/studtmp-a20b5e07f9e90702fdc31c5ad3893ab0106c17b123fc566f4b8cb270313b
2017/06/08 21:27:30.854372  idle-timeout:        5s
2017/06/08 21:27:30.854376  spool-size:          1024
2017/06/08 21:27:30.854380  harvester-buff-size: 16384
2017/06/08 21:27:30.854384  --- flags ---------
2017/06/08 21:27:30.854388  tail (on-rotation):  false
2017/06/08 21:27:30.854391  log-to-syslog:          false
2017/06/08 21:27:30.854395  quiet:             false
2017/06/08 21:27:30.854442 {
  "network": {
    "servers": [ "localhost:44172" ],
    "ssl ca": "/tmp/studtmp-725b02522f1fc58ec885ec14696ecfb9e3347fce3e66c5506dca5d4e4fca"
  },
  "files": [
    { "paths": [ "/tmp/studtmp-42b94bfe144419af867e5284ce7142a9e5e65a55c353eab089a1d5a0f07a" ] }
  ]
}
```

...

```
 1) Logstash-Forwarder TLS Server Verification self signed certificate successfully send the events
     Failure/Error: try(50) { expect(queue.size).to eq(number_of_events), "Expected: #{number_of_events} got: #{queue.size}, execution output:\n #{@execution_output}" }
     
       Expected: 5 got: 0, execution output:
        2017/06/08 21:27:30.854316 	--- options -------
       2017/06/08 21:27:30.854361 	config-arg:          /tmp/studtmp-a20b5e07f9e90702fdc31c5ad3893ab0106c17b123fc566f4b8cb270313b
       2017/06/08 21:27:30.854372 	idle-timeout:        5s
       2017/06/08 21:27:30.854376 	spool-size:          1024
       2017/06/08 21:27:30.854380 	harvester-buff-size: 16384
       2017/06/08 21:27:30.854384 	--- flags ---------
       2017/06/08 21:27:30.854388 	tail (on-rotation):  false
       2017/06/08 21:27:30.854391 	log-to-syslog:          false
       2017/06/08 21:27:30.854395 	quiet:             false
       2017/06/08 21:27:30.854442 {
         "network": {
           "servers": [ "localhost:44172" ],
           "ssl ca": "/tmp/studtmp-725b02522f1fc58ec885ec14696ecfb9e3347fce3e66c5506dca5d4e4fca"
         },
         "files": [
           { "paths": [ "/tmp/studtmp-42b94bfe144419af867e5284ce7142a9e5e65a55c353eab089a1d5a0f07a" ] }
         ]
       }
     
       2017/06/08 21:27:30.854710 Waiting for 1 prospectors to initialise
       2017/06/08 21:27:30.854759 Launching harvester on new file: /tmp/studtmp-42b94bfe144419af867e5284ce7142a9e5e65a55c353eab089a1d5a0f07a
       2017/06/08 21:27:30.854826 harvest: "/tmp/studtmp-42b94bfe144419af867e5284ce7142a9e5e65a55c353eab089a1d5a0f07a" (offset snapshot:0)
       2017/06/08 21:27:30.854873 All prospectors initialised with 0 states to persist
       2017/06/08 21:27:30.855226 Setting trusted CA from file: /tmp/studtmp-725b02522f1fc58ec885ec14696ecfb9e3347fce3e66c5506dca5d4e4fca
       2017/06/08 21:27:30.856198 Connecting to [127.0.0.1]:44172 (localhost) 
       2017/06/08 21:27:30.856328 Failure connecting to 127.0.0.1: dial tcp 127.0.0.1:44172: connection refused
     Shared Example Group: "send events" called from ./spec/integration/logstash_forwarder_spec.rb:91
     # ./spec/support/integration_shared_context.rb:6:in `(root)'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/stud-0.0.22/lib/stud/try.rb:79:in `try'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/stud-0.0.22/lib/stud/try.rb:95:in `try'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/stud-0.0.22/lib/stud/try.rb:91:in `try'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/stud-0.0.22/lib/stud/try.rb:123:in `try'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/logstash-devutils-1.3.3-java/lib/logstash/devutils/rspec/logstash_helpers.rb:18:in `try'
     # ./spec/support/integration_shared_context.rb:6:in `(root)'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
Finished in 8 minutes 47 seconds (files took 3.41 seconds to load)
17 examples, 1 failure, 1 pending
```

I was able to reproduce this one time locally, but dozens of other attempts on multiple platforms worked without issue.  

See Also #219 

Travis: https://travis-ci.org/logstash-plugins/logstash-input-beats/jobs/240959762
Copy: https://gist.github.com/jakelandis/2c51590250b9d9dc798e60d5848bb76d

**Comment by jakelandis** [comment]
I _think_ I found the culprit ...

It seems that when we spawn the child process in the integration tests for logstash forwarder and file beats we only allow 0.1 seconds for that child process to start.  I was able to reliably reproduce this locally by dropping that value down to .0001 seconds. 

The fix here should be to block until the process is started or some max threshold (~5 seconds) is met.
